### PR TITLE
Fix sensor/schedule state in the left nav being outdated

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

In a previous PR, we stopped storing the RootWorkspaceQuery in the Apollo cache. However, this change affected the left navigation, which relied on data from the workspace context. As a result, the left navigation stopped receiving updates to the schedule/sensor state after mutations, since its data source was no longer tied to the Apollo cache.

To resolve this issue, I updated the left navigation to query for the latest status separately. This new query is stored in the Apollo cache, ensuring that any changes to the state on the sensor page or the left navigation are reflected in both places.

## How I Tested These Changes

https://github.com/dagster-io/dagster/assets/2286579/39200fb1-ed51-4620-b725-66b3893d1c02

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/schedules/ScheduleSwitch.tsx [hunk 1]
```diff
@@ -22,18 +26,32 @@ interface Props {
...
+  const scheduleSelector = {
+    ...repoAddressToSelector(repoAddress),
+    scheduleName: name,
+  };
+
+  const {data, loading} = useQuery<ScheduleStateQuery, ScheduleStateQueryVariables>(
+    SCHEDULE_STATE_QUERY,
+    {
+      variables: {scheduleSelector},
+      skip: !hasOutdatedScheduleState,
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/schedules/ScheduleSwitch.tsx:51]: This ScheduleSwitch component is used by places that don't supply outdated data so I added this prop to make the component fetch up to date data only in the left-nav case.

File: js_modules/dagster-ui/package.json [hunk 2]
```diff
@@ -24,6 +24,6 @@
   },
   "packageManager": "yarn@4.1.1",
   "engines": {
-    "node": "21.x"
+    "node": ">=20.x"
```
Comment (salazarm) [inline js_modules/dagster-ui/package.json:27]: This is for storybook builds to pass in vercel which doesnt support node 21

File: js_modules/dagster-ui/packages/ui-core/src/schedules/ScheduleSwitch.tsx (hunk 1 above)
Comment (alangenfeld) [reply js_modules/dagster-ui/packages/ui-core/src/schedules/ScheduleSwitch.tsx:51]: `hasOutdatedScheduleState` -> `shouldFetchLatestState` / `refreshState` / `queryState` or something

"has outdated" miscommunicates that we know its outdated versus opting in to a behavior that resolves the potential problem of being outdated

**Review by alangenfeld — APPROVED** [review]
should we be using this https://www.apollographql.com/docs/react/caching/cache-interaction#watchfragment for a more systemic solution?

**Comment by salazarm** [comment]
> should we be using this https://www.apollographql.com/docs/react/caching/cache-interaction#watchfragment for a more systemic solution?

`useQuery` already has that built in which is what we're relying on.

**Pull request merged**